src/typing_extensions.py

```diff
@@ -145,27 +145,6 @@ def __repr__(self):
 _marker = _Sentinel()
 
 
-def _check_generic(cls, parameters, elen=_marker):
-    """Check correct count for parameters of a generic cls (internal helper).
-    This gives a nice error message in case of count mismatch.
-    """
-    if not elen:
-        raise TypeError(f"{cls} is not a generic class")
-    if elen is _marker:
-        if not hasattr(cls, "__parameters__") or not cls.__parameters__:
-            raise TypeError(f"{cls} is not a generic class")
-        elen = len(cls.__parameters__)
-    alen = len(parameters)
-    if alen != elen:
-        if hasattr(cls, "__parameters__"):
-            parameters = [p for p in cls.__parameters__ if not _is_unpack(p)]
-            num_tv_tuples = sum(isinstance(p, TypeVarTuple) for p in parameters)
-            if (num_tv_tuples > 0) and (alen >= elen - num_tv_tuples):
-                return
-        raise TypeError(f"Too {'many' if alen > elen else 'few'} parameters for {cls};"
```

Comment:
Was trying to figure out why the tests failed when I reuse this error and it is because typing_extensions raises with "parameters" whereas Python has always raised with "arguments", and this conditional is placed everywhere in tests:

https://github.com/python/typing_extensions/blob/fda0c15e3b5ed05703420cfb4d0974edb5e39c46/src/test_typing_extensions.py#L5234

In Python 3.12:

https://github.com/python/cpython/blob/cf29a2f25e359790c0c0f6a4ab0ab49ca5ffab5d/Lib/typing.py#L297

In Python 3.10:

https://github.com/python/cpython/blob/dcb16c98be61630369227f0d893f8d9262d25cac/Lib/typing.py#L231

Is this phrase change on purpose and should the tests be fixed accordingly?